### PR TITLE
Throw error using --context-dir with a template

### LINKED ISSUE #14960: oc new-app --context-dir doesn't work on templates
Similar to #14906, `oc new-app --context-dir=...` parameter is not passed to the builder if a template is being used. 

##### Version
openshift v3.6.121
kubernetes v1.6.1+5115d708d7

##### Steps To Reproduce
`oc new-app -f=https://raw.githubusercontent.com/radanalyticsio/oshinko-s2i/master/scala/scalabuilddc.json -p GIT_URI=https://github.com/pdmack/Spark-MLlib-Movie-Recommendation-JDG-Example.git -p APP_MAIN_CLASS=com.example.data.analytics.App --context-dir=recommend-mllib -o yaml | grep -C 2 recommend-mllib`

No results returned.

##### Current Result
context-dir parameter is not passed to underlying build.

##### Expected Result
Underlying build phase receives and processes change to specified context-dir.

##### Additional Information
new-app example above is incomplete and only sufficient for illustration purposes only.

### PR DESCRIPTION
**Pull request by coreydaley**
--context-dir is not supported when using a template with oc new-app

Fixes #14960

### CODE REVIEW

**Comment by coreydaley** [comment]
[test][testextended][extended:core(builds)]

**Comment by coreydaley** [comment]
@bparees ptal

File: pkg/generate/app/cmd/newapp.go [hunk 1]
```diff
@@ -777,7 +777,9 @@ func (c *AppConfig) Run() (*AppResult, error) {
 	if len(components.UseSource()) > 1 && len(c.To) > 0 {
 		return nil, errors.New("only one component with source can be used when specifying an output image reference")
 	}
-
+	if len(c.ContextDir) > 0 && len(c.TemplateFiles) > 0 {
+		return nil, errors.New("--context-dir is not supported when using a template")
+	}
```
Comment (bparees) [inline pkg/generate/app/cmd/newapp.go]: just to confirm, c.TemplateFiles is populated for all of the following conditions:

1) "oc new-app mylocaltemplate.yaml"
2) "oc new-app -f mylocaltemplate.yaml"
3) "oc new-app http://myremote/template.yaml"
4) "oc new-app -f http://myremote/template.yaml"
5) "oc new-app myclustertemplate"

?

Comment (coreydaley) [inline pkg/generate/app/cmd/newapp.go]: Doesn't look like it, fixed in next push

**Comment by bparees** [comment]
what does this actually output when a user hits it?

**Comment by coreydaley** [comment]
@bparees --context-dir is not supported when using a template

**Comment by bparees** [comment]
@coreydaley i mean the literal output.  i want to see what the user is going to see in terms of full output based on where you're throwing the error.

**Comment by coreydaley** [comment]
@bparees https://pastebin.com/nM50wCjd

**Comment by coreydaley** [comment]
@bparees 

[09:49] $ oc new-app rails-postgresql.json --context-dir=blah 
error: --context-dir is not supported when using a template

**Comment by bparees** [comment]
lgtm, please add a TC to newapp.sh

**Comment by coreydaley** [comment]
test/flake #14575

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/15059
[merge]

**Comment by bparees** [comment]
[merge]

**Pull request merged**